### ISSUE TITLE
Use release version instead of arbitrary dates for Glean metrics expiry

### ISSUE THREAD

**Issue by jonalmeida** [mozilla-mobile/firefox-ios#16373]
In Fenix, we moved over to using release versions for the expiry of Glean metrics so that:

1. they always aligned with when a release went out instead of arbitrary dates which will not always be the same as the release date.
1. metrics will not expire on users that haven't updated to the latest version silently.
1. expiring metrics will not affect building/testing older releases ([context](https://bugzilla.mozilla.org/show_bug.cgi?id=1724760)).

You can compare against [the Android pull request][android_pr] that makes this change and use the appropriate way mentioned in [the glean docs][ios_glean_docs] to set the application's version number for Glean to reference.

rocketsroger may still have [the script](https://github.com/mozilla-mobile/fenix/pull/23805#issue-1141892764) they used to automate conversions from dates to release versions, possibly?

[ios_glean_docs]: https://mozilla.github.io/glean/book/language-bindings/ios/ios-build-configuration-options.html#--expire-by-version-integer
[android_pr]: https://github.com/mozilla-mobile/fenix/pull/23805

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-7386)

**Comment by jonalmeida** [comment]
Jan Erik mentioned on Matrix that there is a slight edge case to consider for v9000 (the test flight version).

**Comment by rocketsroger** [comment]
[update_metrics_expiry_to_version_1.py.zip](https://github.com/mozilla-mobile/firefox-ios/files/12665468/update_metrics_expiry_to_version_1.py.zip)

This is the script I used to convert to version expiry.  This script might need some updating.

**Comment by badboy** [comment]
> Jan Erik mentioned on Matrix that there is a slight edge case to consider for v9000 (the test flight version).

Yup, I don't know how that version is set, but if that's a global flag and it would also be posted to inform Glean about it basically all metrics would always be expired in such a build and we won't get _any_ data (plus tests will fail)
